app/pages/project/workflow-selection.jsx

```diff
@@ -133,16 +104,9 @@ class WorkflowSelection extends React.Component {
             this.getSelectedWorkflow(this.props);
```

Comment:
Should this be replaced with `actions.classifier.loadWorkflow(default_workflow, locale, preferences)`? Loading the default workflow on error seems more sensible to me (as long as we indicate to the user that the workflow has been changed from the one that they thought was loading.)

Reply:
That's a change in behavior that I think we should discuss and if we go forward with it, implement in a separate PR. Even though a lot of projects have default workflows set, not all do, so for what it's worth, I think just running through the strategies again as it does now is overall safer. 

Informing the user is something that @shaunanoordin has brought up and I think better fits as part of the rewrite.

Reply:
Looking at the code, this line only runs if `project.configuration` exists but the selected workflow ID is not the project default. There's no check to see if a project default actually does exist so changing this would introduce bugs.

If there is a workflow loaded at this point, then `actions.classifier.reset()` later in this block will also force selection to run again, since changing the stored workflow unmounts and remounts workflow selection (and the classifier.)

Reply:
These nested if blocks are also very confusing.